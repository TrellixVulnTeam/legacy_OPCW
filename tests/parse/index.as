# Declare an array of 20 ints.
let a: int[20];

# Utilize the array.
a[2];
a[5] = 3420;

# Declare an array of 10 tuples.
let b: (int, bool)[10];

# Utilize a crazy array.
(d)[4][41];
d[a[1]][a[2]];

# Declare an array of 10 things.
struct Thing { }
let t: Thing[5 * 2];

# Declare a blob with a matrix.
struct Blob { m: uint[5][5] }
let b: Blob[12];
