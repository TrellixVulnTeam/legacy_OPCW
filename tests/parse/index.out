ModuleDecl <?> _
  Nodes <?> 
    SlotDecl <?> a
      ArrayType <?>
        IntegerExpr <?> 20 (10)
        Ident <?> int
    IndexExpr <?>
      Ident <?> a
      IntegerExpr <?> 2 (10)
    AssignExpr <?>
      IndexExpr <?>
        Ident <?> a
        IntegerExpr <?> 5 (10)
      IntegerExpr <?> 3420 (10)
    SlotDecl <?> b
      ArrayType <?>
        IntegerExpr <?> 10 (10)
        TupleType <?>
          Elements <?> 
            TupleTypeMem <?>
              Ident <?> int
            TupleTypeMem <?>
              Ident <?> bool
    IndexExpr <?>
      IndexExpr <?>
        Ident <?> d
        IntegerExpr <?> 4 (10)
      IntegerExpr <?> 41 (10)
    IndexExpr <?>
      IndexExpr <?>
        Ident <?> d
        IndexExpr <?>
          Ident <?> a
          IntegerExpr <?> 1 (10)
      IndexExpr <?>
        Ident <?> a
        IntegerExpr <?> 2 (10)
    Struct <?> Thing
      Type Parameters <?> 
      Members <?> 
    SlotDecl <?> t
      ArrayType <?>
        MultiplyExpr <?>
          IntegerExpr <?> 5 (10)
          IntegerExpr <?> 2 (10)
        Ident <?> Thing
    Struct <?> Blob
      Type Parameters <?> 
      Members <?> 
        StructMem <?> m
          ArrayType <?>
            IntegerExpr <?> 5 (10)
            ArrayType <?>
              IntegerExpr <?> 5 (10)
              Ident <?> uint
    SlotDecl <?> b
      ArrayType <?>
        IntegerExpr <?> 12 (10)
        Ident <?> Blob
